### PR TITLE
[Bugfix] Fixes TA grading status page error

### LINKED ISSUE #4201: Error when trying to enter TA grading
Frog robot 
Not sure why this happens in one course on one gradeable, but not on other courses/gradeables.

17:09:27 07/25/2019 - FATAL ERROR
TypeError (Code: 0) thrown in /usr/local/submitty/site/app/views/grading/ElectronicGraderView.php (Line 33) by:
    public function statusPage(

Message:
Argument 8 passed to app\views\grading\ElectronicGraderView::statusPage() must be of the type integer, null given

Stack Trace:
#0 unknown file(unknown line): app\views\grading\ElectronicGraderView->statusPage(app\models\gradeable\Gradeable, Array, Array, NULL, NULL, NULL, 0,
 NULL, 0, 0, 0, 0, 'registration_section', 0, true)
#1 /usr/local/submitty/site/app/libraries/Output.php(193): call_user_func_array(Array, Array)
#2 unknown file(unknown line): app\libraries\Output->renderTemplate(Array, 'statusPage', app\models\gradeable\Gradeable, Array, Array, NULL, NULL, N
ULL, 0, NULL, 0, 0, 0, 0, 'registration_section', 0, true)
#3 /usr/local/submitty/site/app/libraries/Output.php(158): call_user_func_array('self::renderTemplate', Array)
#4 /usr/local/submitty/site/app/controllers/grading/ElectronicGraderController.php(410): app\libraries\Output->renderOutput(Array, 'statusPage', app
\models\gradeable\Gradeable, Array, Array, NULL, NULL, NULL, 0, NULL, 0, 0, 0, 0, 'registration_section', 0, true)
#5 unknown file(unknown line): app\controllers\grading\ElectronicGraderController->showStatus('x')
#6 /usr/local/submitty/site/app/libraries/routers/WebRouter.php(157): call_user_func_array(Array, Array)
#7 /usr/local/submitty/site/app/libraries/routers/WebRouter.php(138): app\libraries\routers\WebRouter->run()
#8 /usr/local/submitty/site/public/index.php(339): app\libraries\routers\WebRouter::getWebResponse(Symfony\Component\HttpFoundation\Request, app\lib
raries\Core, true)

URL: http://192.168.56.111//f19/a_b/gradeable/x/grading/status

### PR DESCRIPTION
**Pull request by zjxiaohan**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
Fixes #4201

### CODE REVIEW

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 1]
```diff
@@ -405,7 +405,7 @@ public function showStatus($gradeable_id) {
         }
 
         $this->core->getOutput()->renderOutput(array('grading', 'ElectronicGrader'), 'statusPage', $gradeable, $sections,
-            $component_averages, $autograded_average, $overall_scores, $overall_average, $total_submissions, $individual_viewed_grade,
+            $component_averages, $autograded_average, $overall_scores, $overall_average, $total_submissions, $individual_viewed_grade ?? 0,
```
Comment (zjxiaohan) [inline site/app/controllers/grading/ElectronicGraderController.php]: https://github.com/Submitty/Submitty/commit/ce6a9e5448c7654aa8ec8e15d3ea3252012d9a67

File: site/app/controllers/student/SubmissionController.php [hunk 2]
```diff
@@ -157,15 +157,15 @@ public function showHomeworkPage($gradeable_id, $gradeable_version = null) {
                 // Only show hidden test cases if the display version is the graded version (and grades are released)
                 $show_hidden = false;
                 if ($graded_gradeable != NULL) {
-                  $show_hidden = $version == $graded_gradeable->getOrCreateTaGradedGradeable()->getGradedVersion(false) && $gradeable->isTaGradeReleased();
+                    $show_hidden = $version == $graded_gradeable->getOrCreateTaGradedGradeable()->getGradedVersion(false) && $gradeable->isTaGradeReleased();
+                    $can_inquiry = $this->core->getAccess()->canI("grading.electronic.grade_inquiry", ['graded_gradeable' => $graded_gradeable]);
```
Comment (zjxiaohan) [inline site/app/controllers/student/SubmissionController.php]: https://github.com/Submitty/Submitty/commit/477eb4aafc742c9452c1941f5c744ab804b554a4

File: site/app/views/submission/HomeworkView.php [hunk 3]
```diff
@@ -397,7 +397,7 @@ private function renderSubmitBox(Gradeable $gradeable, $graded_gradeable, $versi
             'component_names' => $component_names,
             'upload_message' => $this->core->getConfig()->getUploadMessage(),
             "csrf_token" => $this->core->getCsrfToken(),
-            'has_overridden_grades' => $graded_gradeable->hasOverriddenGrades(),
+            'has_overridden_grades' => $graded_gradeable ? $graded_gradeable->hasOverriddenGrades() : false,
```
Comment (zjxiaohan) [inline site/app/views/submission/HomeworkView.php]: https://github.com/Submitty/Submitty/commit/036bd71255dcde68a0cdd940ed099ea3eb1da386

File: site/app/controllers/student/SubmissionController.php (hunk 2 above)
Comment (bmcutler) [reply site/app/controllers/student/SubmissionController.php]: what changed on this line?

Comment (zjxiaohan) [reply site/app/controllers/student/SubmissionController.php]: ` $can_inquiry = $this->core->getAccess()->canI("grading.electronic.grade_inquiry", ['graded_gradeable' => $graded_gradeable]);` moved inside the `if`.

`graded_gradeable` doesn't always exist.

Comment (bmcutler) [reply site/app/controllers/student/SubmissionController.php]: Sorry, nevermind, I was just looking at the indentation of the first line.

**Pull request merged**